src/agent/ipc/ipc_workers/tcp_worker_impl.cc

```diff
@@ -188,8 +188,40 @@ SpTcpWorker::Inject(SpChannel* c,
 
   sp_debug("INJECT CMD -- %s", cmd_exe.c_str());
 
+  // remove agent library from the LD_PRELOAD environment variable
+  // in the case that agent library is injected through LD_PRELOAD
+  char* ld_preload_str = std::getenv("LD_PRELOAD");
+
+  if (ld_preload_str == NULL) {
```

Comment:
While C++ certainly doesn't have great string manipulation capabilities, we can try to keep our sanity by staying away from `strtok` and its ilk.

    if(char* ld_preload = std::getenv("LD_PRELOAD")) {
        std::string libs{ld_preload};

        // Find the location where the agent name starts
        auto const start = libs.find(agent_name);
        if(start != std::string::npos) {
            // Find the previous separator
            auto prev = libs.rfind(':', start);

            // If none, then start from the beginning of the string
            if(prev == std::string::npos) prev = 0;

            // remove the agent library name
            libs.replace(prev, start - prev + agent_name.length(), "");

            // Replace the env. variable (NB: this requires a POSIX system)
            setenv("LD_PRELOAD", libs.c_str(), true);
        }
    }

Reply:
Thank you. LD_PRELOAD has two possible delimiters, space or colon, according to the doc. I didn't find a std::string function that does rfind with two delimiters, so I just call string::rfind twice and compare the results.

Reply:
Oh, good catch. It could be done with `std::find` and reverse iterators, but this solution works just as well.